lib/metadataTypes/User.js

```diff
@@ -0,0 +1,930 @@
+        /** @type {TYPE.UserConfiguration[]} */
+        const configs = [];
+        for (const action in this.userBUassignments) {
+            for (const data of Object.entries(this.userBUassignments[action])) {
+                const buIds = data[1];
+                let userId = data[0];
+                if (!userId) {
+                    continue;
+                }
+                userId = userId.startsWith('key:') ? upsertResults[userId.slice(4)].ID : userId;
+                configs.push(
+                    /** @type {TYPE.UserConfiguration} */ {
+                        Client: { ID: this.buObject.eid },
+                        ID: userId,
+                        BusinessUnitAssignmentConfiguration: {
+                            BusinessUnitIds: { BusinessUnitId: buIds },
+                            IsDelete: action === 'delete',
+                        },
+                    }
+                );
+            }
+        }
+        if (configs.length > 0) {
+            Util.logger.info('Deploying: BU assignment changes');
+            // run update
+            const buResponse = await this.client.soap.configure('AccountUser', configs);
+            // process response
+            if (buResponse.OverallStatus === 'OK') {
+                // get userIdNameMap
+                const userIdNameMap = {};
+                for (const user of Object.values(upsertResults)) {
+                    userIdNameMap[user.ID] = `${user.CustomerKey} / ${user.Name}`;
+                }
+                // log what was added / removed
+                let configureResults = buResponse.Results?.[0]?.Result;
+                if (!configureResults) {
+                    Util.logger.debug(
+                        'buResponse.Results?.[0]?.Result not defined: ' + JSON.stringify(buResponse)
+                    );
+                    return;
+                }
+                if (!Array.isArray(configureResults)) {
+                    configureResults = [configureResults];
+                }
+                const userBUresults = {};
+                for (const result of configureResults) {
+                    if (result.StatusCode === 'OK') {
+                        /** @type {TYPE.UserConfiguration} */
+                        const config = result.Object;
+                        const buArr =
+                            config.BusinessUnitAssignmentConfiguration.BusinessUnitIds
+                                .BusinessUnitId;
+                        userBUresults[config.ID] = userBUresults[config.ID] || {
+                            add: [],
+                            delete: [],
+                        };
+                        userBUresults[config.ID][
+                            config.BusinessUnitAssignmentConfiguration.IsDelete ? 'delete' : 'add'
+                        ] = Array.isArray(buArr) ? buArr : [buArr];
+                    } else {
+                        // TODO: handle (so far unknown) error
```

Comment:
can we at least log something here - otherwise will be one of those needle in haystack thigns

Reply:
done